hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestUpdateConfiguration.java

```diff
@@ -47,31 +49,56 @@
   private static final Logger LOG = LoggerFactory.getLogger(TestUpdateConfiguration.class);
   private final static HBaseTestingUtility TEST_UTIL = new HBaseTestingUtility();
 
+  private static Path newCnfPath, newCnf2Path, newCnf3Path;
+
   @BeforeClass
   public static void setup() throws Exception {
-    // Set master number and use default values for other options.
+    Path cnfPath = FileSystems.getDefault().getPath("target/test-classes/hbase-site.xml");
+    Path cnf2Path = FileSystems.getDefault().getPath("target/test-classes/hbase-site2.xml");
+    Path cnf3Path = FileSystems.getDefault().getPath("target/test-classes/hbase-site3.xml");
+
+    String absoluteDataPath = TEST_UTIL.getDataTestDir().toString();
+    String dataBasePath = System.getProperty(HBaseCommonTestingUtility.BASE_TEST_DIRECTORY_KEY,
+      HBaseCommonTestingUtility.DEFAULT_BASE_TEST_DIRECTORY);
+
+    String dataPath = absoluteDataPath.substring(absoluteDataPath.indexOf(dataBasePath));
+    newCnfPath = Paths.get(dataPath + "/hbase-site.xml");
+    newCnf2Path = Paths.get(dataPath + "/hbase-site2.xml");
+    newCnf3Path = Paths.get(dataPath + "/hbase-site3.xml");
+
+    // Copy Files from target/test-classes to data directory, this is to avoid messing up
+    // the config file under target/test-class, which is supposed to be immutable.
+    Files.createDirectories(Paths.get(dataPath));
+    Files.copy(cnfPath, newCnfPath);
+    Files.copy(cnf2Path, newCnf2Path);
+    Files.copy(cnf3Path, newCnf3Path);
```

Comment:
Duped code. Any way of sharing ?

Reply:
Thought about it, let me come back to see if there is good way to share code between these two testing classes.

Reply:
Will leave it as it is now, do not look that bad after changing to loop, :)

Reply:
Sweet